src/tools/immutable_object_cache/ObjectCacheStore.cc

```diff
@@ -302,5 +356,74 @@ std::string ObjectCacheStore::get_cache_file_path(std::string cache_file_name,
   return m_cache_root_dir + cache_file_dir + cache_file_name;
 }
 
+void ObjectCacheStore::handle_throttle_ready(void* ctx, uint64_t flag) {
+  m_io_throttled.store(false);
+}
+
+bool ObjectCacheStore::take_token_from_throttle(uint64_t object_size,
+                                                uint64_t object_num) {
+  bool wait = false;
+
+  if (m_io_throttled.load() == true) {
+    return false;
+  }
+
+  if (!wait && (m_qos_enabled_flag & ROC_QOS_IOPS_THROTTLE)) {
```

Comment:
Nit: you can probably just implement this as a loop over `m_throttles` so that you can re-use the same code

Reply:
> Nit: you can probably just implement this as a loop over `m_throttles` so that you can re-use the same code

If Iuse loops, these two sentences get more complicated because there are two sets of difference values.
(ROC_QOS_IOPS_THROTTLE, object_num)
(ROC_QOS_BPS_THROTTLE, object_size )

Reply:
Fair enough